src/runtime/patch/process.js

```diff
@@ -0,0 +1,45 @@
+export function patchProcess(sfsRuntime) {
+  // hot-patch process.exit so when it's called we shutdown the patcher early.
+  // we can't just use the event because it's not early enough
+  const process_exit = process.exit;
+
+  // setup exit fn
+  // inspired on: // https://github.com/sindresorhus/exit-hook
+  let exitWasCalled = false;
+  const exitFn = (n) => {
+    if (!exitWasCalled) {
+      exitWasCalled = true;
+
+      // unlocks the files.
+      sfsRuntime.staticfilesystem.shutdown();
+
+      // remove and undo all the patches
+      sfsRuntime.undo();
+    }
+
+    process_exit(n);
+  };
+
+  // apply patches
+  // main exit patch
+  process.exit = (n) => {
+    exitFn(n);
+  };
+
+  // special patch for pm2 cluster shutdown message
```

Comment:
Do we support running Kibana under `pm2`?

Reply:
I don't think we officially support it but there is some users across our discuss forum that look like are using it. In that way I choose to anticipate that!